### ISSUE TITLE
--enable-rpath gives up even if RUNPATH is set

### ISSUE THREAD

**Issue by mnmelo** [plumed/plumed2#807]
Hello,
I'm compiling PLUMED 2.8.0 on Ubuntu 18.04 with g++ 7.3.0 (and readelf v2.30).
`configure` won't accept `--enable-rpath`:
```
configure: WARNING: -R option seems not working, disabling rpath
```
because in my system the compiled test object get its `RUNPATH` set, instead of `RPATH`. `readelf -d` yields:
```
...
0x000000000000001d (RUNPATH)            Library runpath: [/some/random/dir/]
...
```
which fails `configure`'s grep for `RPATH`.

I don't know why `RUNPATH` is generated instead of `RPATH`, but should serve the purpose just as well, and I guess the grep in `configure` could be updated to accept either.

Am I missing something?

**Comment by mnmelo** [comment]
Additionally, even working around the `configure` issue, only the libraries (`libplumed.so`, `libplumedKernel.so`) get `RUNPATH` set, but not the `plumed` executable.

**Comment by GiovanniBussi** [comment]
@mnmelo the original reason for this flag was to hard code the path of libraries plumed depends on. I mean, in a `module load` context:
- I use `module load xdrfile`, which sets `LD_LIBRARY_PATH`
- I compile plumed
- When I run plumed, I don't need to load `xdrfile` again.

In this context, it makes sense to:
- only use the rpath flags when linking plumed libraries (and not the executable)
- only include `LD_LIBRARY_PATH` in the rpath

Starting with v2.8, it [was asked](https://github.com/plumed/plumed2/pull/767) to also add `$libdir` to the saved path.

Based on your comment, it could make more sense to use this flag also when linking the `plumed` executable, so that it can find the plumed library.

To solve your practical problem, you can just pass the proper flags to both `LIBS` (to be used with the libraries) and to `STATIC_LIBS` (to be used with the executable).

Can you check if this would work in your case?

If so we can think how to fix this in configure.ac

**Comment by mnmelo** [comment]
Thanks for the feedback, @GiovanniBussi
I confirm that I can force my own `RPATH`/`RUNPATH` using `LIBS` or `STATIC_LIBS`. I'll be using those flags to precisely add `$libdir` to the `RPATH`/`RUNPATH` and so an `--enable-rpath` that also took care of the `plumed` executable would simplify a bit my compilation scripts.

Thanks a lot already for the info, even if the behavior of  `--enable-rpath` isn't extended.

**Comment by GiovanniBussi** [comment]
Great! I keep this in todo, so leave this issue open. thanks for reporting

**Comment by GiovanniBussi** [comment]
@mnmelo could you please check if branch v2.8-fix-807 solves the problem? As soon as you confirm, I will merge this in v2.8. Thanks!

**Comment by mnmelo** [comment]
Thanks a lot!
I can't try it right now, but just from code inspection it seems you forgot to actually change [the grep for the `RUNPATH` case](https://github.com/plumed/plumed2/blob/575fbaa7a322d9ac0d23a5acff61b94f65126417/configure.ac#L972).

Also, there is now [different `LD_MAIN_RUNPATH` behavior between the `RPATH` and `RUNPATH` cases](https://github.com/plumed/plumed2/blob/575fbaa7a322d9ac0d23a5acff61b94f65126417/configure.ac#L974). Should it be the same?

**Comment by GiovanniBussi** [comment]
Sorry stupid mistake here... should be fixed now

**Comment by GiovanniBussi** [comment]
Fixed in 395deb7e79d57393546b3a2ae7969116471bfebb
